### PR TITLE
Transferring infra inventory and key management

### PR DESCRIPTION
**Pull request by acozine**
We are retiring the old `infra` repository and consolidating automation in this repo. This PR:

- adds inventory entries that were tracked in `infra` but not here
  - CDH machines
  - google cloud machines
  - more aws machines
  - sandbox machines
  - some other stuff (may be out of date)
- adds key management tasks  for the `pulsys` user
  - adds a pulsys user keys playbook and template
  - groups library, CDH, and other keys in the authorized_keys template
  - adds group_vars files that apply only to specific inventory groups so we can manage access flexibly

### CODE REVIEW

File: group_vars/cdh/common.yml
Comment (kayiwa) [inline group_vars/cdh/common.yml:1]: 😎

File: inventory/all_projects/confluence [hunk 1]
```diff
@@ -1,3 +1,5 @@
 [confluence_staging]
 lib-confluence-staging1.princeton.edu
-# do we have a production confluence vm?
\ No newline at end of file
+[confluence_production]
+lib-confluence-prod1.princeton.edu
+confluence_prod1.princeton.edu
```
Comment (kayiwa) [inline inventory/all_projects/confluence:5]:
```suggestion
confluence-prod1.princeton.edu
```

**Review by kayiwa — COMMENTED** [review]
Two jumped out otherwise LGTM

**Comment by acozine** [comment]
Sigh. This doesn't do what we want yet. See https://github.com/PrincetonUniversityLibrary/infra/issues/42 for details. I'm converting this to a draft and will pick it up again next week.

Testing on a sub-group of servers including CDH, EALapps, and FPUL. With all the `when` restrictions and `run_once` on the keys tasks, the first server must be one that takes all of the keys, or the steps get skipped. Maybe my conditions (`when`) are wrong. Maybe there's a better way to do this (some kind of variable concatenation?). Will pick this up again next week.

File: group_vars/all/vars.yml [hunk 2]
```diff
@@ -69,6 +69,42 @@ project_db_host: http://127.0.0.1
...
+  - https://github.com/acozine.keys
+  - https://github.com/gpmenos.keys
+  - https://github.com/kayiwa.keys
+  - https://github.com/leefaisonr.keys
+  - https://github.com/VickieKarasic.keys
+cdh_github_keys:
+  - https://github.com/blms.keys # Ben Silverman, external consultant
+  - https://github.com/quadrismegistus.keys
+  - https://github.com/rlskoeser.keys
+eal_github_keys:
```
Comment (kayiwa) [inline group_vars/all/vars.yml:106]: snowflake

File: inventory/all_projects/_orphans [hunk 3]
```diff
@@ -1,8 +1,11 @@
 # alphabetized by group name
 [geoserver]
 geoserv1.princeton.edu
+[issue_tracker]
+issuetracker-prod1.princeton.edu
 [monitor_backends]
 lib-monitor1.princeton.edu
+sensu-testing-staging-1.princeton.edu # the "punching bag" VM
```
Comment (kayiwa) [inline inventory/all_projects/_orphans:8]: :-)

File: playbooks/pulsys_user.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - name: get library staff keys from github
+      ansible.builtin.command: 'curl {{ item }}'
+      loop: "{{ library_github_keys }}"
+      register: library_keys_from_github
+      run_once: true
+
+    - name: omit library keys from CDH boxes
+      ansible.builtin.set_fact:
+        library_keys_from_github: []
+      when: "'cdh' in inventory_hostname"
```
Comment (kayiwa) [inline playbooks/pulsys_user.yml:29]: COOL!

**Comment by kayiwa** [comment]
OK! THANKS SO MUCH @acozine

**Pull request merged**